config/deploy/docker/lib/aws_sdk_helpers.rb

```diff
@@ -0,0 +1,105 @@
+require 'aws-sdk-iam'
+require 'aws-sdk-elasticloadbalancingv2'
+require 'aws-sdk-ecr'
+require 'aws-sdk-ecs'
+require 'aws-sdk-secretsmanager'
+require 'aws-sdk-autoscaling'
+require 'aws-sdk-ec2'
+require 'aws-sdk-cloudwatchevents'
+require 'aws-sdk-cloudwatch'
+require 'aws-sdk-cloudwatchlogs'
+
+module AwsSdkHelpers
+  module ClientMethods
+    define_singleton_method(:iam) { Aws::IAM::Client.new }
+    define_singleton_method(:elbv2) { Aws::ElasticLoadBalancingV2::Client.new }
+    define_singleton_method(:ecr) { Aws::ECR::Client.new }
+    define_singleton_method(:ecs) { Aws::ECS::Client.new }
+    define_singleton_method(:secretsmanager) { Aws::SecretsManager::Client.new }
+    define_singleton_method(:autoscaling) { Aws::AutoScaling::Client.new}
+    define_singleton_method(:ec2) { Aws::EC2::Client.new}
+    define_singleton_method(:cloudwatchevents) { Aws::CloudWatchEvents::Client.new }
+    define_singleton_method(:cw)  { Aws::CloudWatch::Client.new }
+    def self.cwl(profile = nil)
+      profile ||= AwsSdkHelpers::Helpers.cluster_name
+      Aws::CloudWatchLogs::Client.new(profile: profile)
+    end
+
+    define_method(:iam)              { AwsSdkHelpers::ClientMethods.iam }
+    define_method(:elbv2)            { AwsSdkHelpers::ClientMethods.elbv2 }
+    define_method(:ecr)              { AwsSdkHelpers::ClientMethods.ecr }
+    define_method(:ecs)              { AwsSdkHelpers::ClientMethods.ecs }
+    define_method(:secretsmanager)   { AwsSdkHelpers::ClientMethods.secretsmanager }
+    define_method(:autoscaling)      { AwsSdkHelpers::ClientMethods.autoscaling }
+    define_method(:ec2)              { AwsSdkHelpers::ClientMethods.ec2 }
+    define_method(:cloudwatchevents) { AwsSdkHelpers::ClientMethods.cloudwatchevents }
+    define_method(:cw)               { AwsSdkHelpers::ClientMethods.cw }
+    define_method(:cwl)              { AwsSdkHelpers::ClientMethods.cwl(
+      profile: respond_to?(:_cluster_name) ? _cluster_name : AwsSdkHelpers::Helpers.cluster_name,
+    ) }
+  end
+
+  module Helpers
+    include AwsSdkHelpers::ClientMethods
+
+    def self.cluster_name
+      ENV.fetch('CLUSTER_NAME', ENV.fetch('AWS_CLUSTER', ENV.fetch('AWS_PROFILE', ENV.fetch('AWS_VAULT', ''))))
```

Comment:
We shouldn't propagate my bad defaulting logic if possible. We should just settle on one of these (I like the first).